### ISSUE TITLE
False-positive misc-unused-alias-decls when namespace-alias comes before include

### ISSUE THREAD

**Issue by HighCommander4** [llvm/llvm-project#59907]
```console
$ cat a.hpp
#ifndef A_HPP_
#define A_HPP_

namespace A {
  namespace fs {
    struct path {};
  }
}

#endif
```

```console
$ cat b.hpp
#include "a.hpp"
namespace fs = A::fs;
struct B {
  fs::path member;
};
```

```console
$ cat c.hpp
#include "a.hpp"
namespace fs = A::fs;

#include "b.hpp"

struct C {
  fs::path member1;
  B member2;
};
```

```console
$ clang-tidy --checks=misc-unused-alias-decls c.hpp
Running without flags.
1 warning generated.
/home/botond/temp/clangd-debugging/issue-1438/c.hpp:2:11: warning: namespace alias decl 'fs' is unused [misc-unused-alias-decls]
namespace fs = A::fs;
~~~~~~~~~~^~~~~~~~~~~
```

It seems like a valid stylistic choice to group the namespace-alias with the #include to which it pertains, so this seems like an undesired false-positive to me.
